supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystem.scala

```diff
@@ -198,8 +205,25 @@ trait SharedFileSystem extends PathFactory {
       PairOfFiles(src, dest)
     }
 
+    // A possibly staged version of the input file suitable for downstream processing, or just the original input
+    // file if no staging was required.
+    val staged: WomFile = value.mapFile { input =>
+      pathBuilders.collectFirst({ case h: HttpPathBuilder if HttpPathBuilder.accepts(input) => h }) match {
+        case Some(httpPathBuilder) =>
+          implicit val materializer = ActorMaterializer()
+          implicit val ec = actorContext.dispatcher
+          val stagedFile = better.files.File.newTemporaryFile()
+          val futurePath = httpPathBuilder.contentAsByteIterator(input) map { bytes =>
+            stagedFile.writeBytes(bytes)
+            stagedFile.pathAsString
```

Comment:
ToL: it smells a little fishy to me that it's the Cromwell server itself rather than the target system that's doing the downloading.

eg if 100,000 jobs are submitted we will open 100,000 connections and stream 100,000 files before adding jobs to the SGE queue (for example). If the backend was doing it as part of the job (like in the PAPI2 case), we could leverage the existing queuing and breadth of computational resource to take care of all that

Reply:
It's a good point, OTOH submitting 100,000 SGE jobs is also not going to go so well on most systems

Reply:
Maybe I'm just especially sensitive right now to HTTP-related things that happen "in-JVM" if we don't know how well they scale...